### DISCUSSION TITLE
why is it that add nanobind to my CMake project results in a warning about ASAN not coming first in the library list?

### DISCUSSION THREAD

**Discussion by starball5** [wjakob/nanobind#623] [Q&A] [answered]
I have a CMake project where adding nanobind results in build failing with the message

> ASan runtime does not come first in initial library list; you should either link runtime to your application or manually preload it with LD_PRELOAD

Toolchain:

- GCC 13.2.0
- ninja 1.11.0
- CMake 3.29.5
- nanobind 2.0.0
- sanitizers: address + undefined

**Comment by wjakob** [comment]
That's not a nanobind issue but a more general one involving Python. Either you compile your own Python with ASAN, or you need to use LD_PRELOAD as the message says. If you load an ASAN-based shared library into a non-ASAN process, it's already too late.

**Reply by wjakob** [reply, marked as answer]
This may be useful by the way: it explains how to use sanitizers in one of the projects I'm developing: https://drjit.readthedocs.io/en/latest/debug.html#sanitizing-python-sessions

**Reply by starball5** [reply]
is this now fixed by https://github.com/wjakob/nanobind/pull/1000 ?

**Reply by wjakob** [reply]
Should be!
